Add nav menu field to scf

haseebnawaz298 opened the pull request:
- this is a very common field that is used a lot, and it is not in scf.

haseebnawaz298 committed:
Add nav menu field to scf

kraftbj commented on includes/fields/class-acf-field-nav-menu.php:
```diff
@@ -0,0 +1,207 @@
+ * This is a PHP file containing the code for the acf_field_nav_menu class.
+ *
+ * @package wordpress/secure-custom-fields
+ * @since 6.4.1
```
```suggestion
 * @since 6.5.0
```

kraftbj commented on includes/fields/class-acf-field-nav-menu.php:
```diff
@@ -0,0 +1,207 @@
+		 *
+		 * @type    function
+		 * @date    5/03/2014
+		 * @since   ACF 6.4.1
```
```suggestion
		 * @since  6.5.0
```

kraftbj commented on docs/features/field/nav_menu/tutorial.md:
```diff
@@ -0,0 +1,18 @@
+
+## Common Use Cases
+
+1. Use to output wordpress menu anywhere
```
```suggestion
1. Use to output WordPress navigation menu anywhere.
```

kraftbj reviewed (commented):
This works seems to work well. A couple of questions:

1. Themes without nav menus (e.g. Twenty Twenty Four) will output a notice on the post editor that the theme doesn't support it. I'm curious if there's a better way of handling it (notice on the field group editor? Disable the Nav Menu field when using a theme that doesn't support it? Hide the field on the post editor if the theme doesn't support it, etc.
2. Could you walk me through how you're using this? I'd just like to understand the user journey on it. Something like "this post is about Cats, so I want to display my Animals-specific navigation menu under the top nav" with more flexibility than, say, logic by category?

I mainly want to ensure I'm understanding the need properly.

There are a couple of version updates needed (this will go in 6.5) and tweaking the docs with a typo. [I'm happy to make those edits myself before merge, so no worries there].

With the tutorial, please feel free to be more verbose. The docs, overall, are really spartan but I hope to flesh them out so someone could follow the tutorial from start to finish to add a field, use a field via the post editor, and output the field.

haseebnawaz298 commented:
1. we can go with showing a message in field group editor and hiding the field from post editor, then we donot need to show any notices.
2. I have worked on many sites where this was needed we did use a plugin for it, but that plugin is no longer maintained.

some of the common use case that i normally use is.

-  A second menu that can be set as a global (Theme Options) and also be edited in a page( Page options), if page option is set to be empty global one is fetched.
- it also gives you the ability to show different menus based on different conditions

haseebnawaz298 commented:
I would also love to help with the docs, just need a direction on how deep we want the docs to be. eg: do we need to give code examples etc, or just basic instructions?

kraftbj committed:
Minor edits

kraftbj commented:
> 1. we can go with showing a message in field group editor and hiding the field from post editor, then we donot need to show any notices.

Awesome; are you good with making those changes on this branch before merging? (note that I did commit the suggestions I made in the initial review so please pull before new commits)



> I would also love to help with the docs, just need a direction on how deep we want the docs to be. eg: do we need to give code examples etc, or just basic instructions?

I'd say basic code examples. I'm thinking of two types of users using the docs:
1. The experienced person looking for details. We don't need to do too much beyond good inline documentation of the args, describing the options, etc.
2. Someone new to WordPress/web development who need something more like training wheels.

For the latter person, if they said "I want to add a new field to my site" and stumbled upon SCF, will the tutorial provide them enough information to be able to add the field and output it? Until we have better integration with blocks, that's going to almost always be with small php code.

So, for the nav menu, I would say it could be something like "To output this field, add this code to the theme template, functions.php or elsewhere as needed for implementation `..... get_field(.....)......`"

Then with a small tutorial showing "As an example, using a child theme of the Twenty Twelve default theme, if your nav menu field is "special-menu", add [code] to the such and such template.php"

Does that make sense? Obviously few people would need the specific tutorial exactly as written, but enough where someone could get it work in that generic place and be enough to hopefully connect some dots for them.

Additionally, I am okay with merging this (with the notice update) without the docs being 100% and accepting a follow-up PR for it.

Thank you!

haseebnawaz298 commented:
i will work on this hopefully today or tomorrow, will get back to you soon

haseebnawaz298 committed:
Merge branch 'WordPress:trunk' into trunk

haseebnawaz298 committed:
update will the changes

- hide field in post editor if there is no support
- show notice in field group editor

haseebnawaz298 commented:
This is how the notices show in the group field editor now,

field is hidden if there is no support in theme.

![image](https://github.com/user-attachments/assets/96934c02-63e2-4a59-a8bc-c821692d4304)
![image](https://github.com/user-attachments/assets/6b58022a-3dab-4e94-aaba-c6813f94b07d)

haseebnawaz298 commented:
also updated the docs to be a bit better

haseebnawaz298 committed:
Update tutorial.md

haseebnawaz298 committed:
Merge branch 'WordPress:trunk' into trunk

kraftbj commented on includes/fields/class-acf-field-nav-menu.php:
```diff
@@ -0,0 +1,237 @@
+					<div class="acf-field">
+						<div class="acf-notice">
+							<p>
+								<?php esc_html_e( 'Warning: You don\'t have any menus created please visit', 'secure-custom-fields' ); ?> <a href="<?php echo esc_url( admin_url( 'nav-menus.php' ) ); ?>"><?php esc_html_e( 'this', 'secure-custom-fields' ); ?></a> <?php esc_html_e( 'link to create menus.', 'secure-custom-fields' ); ?>
```
We'd probably want to incorporate the variable into the translation field: https://developer.wordpress.org/plugins/internationalization/how-to-internationalize-your-plugin/#variables

Additionally, can we have the actual link be more descriptive? Instead of "this", something like "This field requires an existing navigation menu, please ==create a menu first==." (`==` to show the linked text)

kraftbj commented on includes/fields/class-acf-field-nav-menu.php:
```diff
@@ -0,0 +1,237 @@
+				<div class="acf-field">
+					<div class="acf-notice">
+						<p>
+							<?php esc_html_e( 'Warning: The theme does not support navigation menus, the field will not show.', 'secure-custom-fields' ); ?>
```
```suggestion
							<?php esc_html_e( 'Warning: The theme does not support navigation menus, the field will not display.', 'secure-custom-fields' ); ?>
```

kraftbj reviewed (changes requested):
I haven't actually tested it yet, but a couple of comments from reviewing the code.

haseebnawaz298 committed:
Update includes/fields/class-acf-field-nav-menu.php

Co-authored-by: Brandon Kraft <public@brandonkraft.com>

haseebnawaz298 committed:
Merge branch 'WordPress:trunk' into trunk

haseebnawaz298 committed:
update no menu message

kraftbj reviewed (approved):
Tests well.

haseebnawaz298 commented:
nice
